go/arrow/array/string.go

```diff
@@ -79,16 +79,12 @@ func (a *String) ValueOffsets() []int32 {
 	return a.offsets[beg:end]
 }
 
-func (a *String) ValueBytes() (ret []byte) {
+func (a *String) ValueBytes() []byte {
 	beg := a.array.data.offset
 	end := beg + a.array.data.length
 	data := a.values[a.offsets[beg]:a.offsets[end]]
 
-	s := (*reflect.SliceHeader)(unsafe.Pointer(&ret))
-	s.Data = (*reflect.StringHeader)(unsafe.Pointer(&data)).Data
-	s.Len = len(data)
-	s.Cap = len(data)
-	return
+	return []byte(data)
```

Comment:
this causes a copy which isn't what we want. This could end up creating a copy of the entire array essentially which would be pretty bad. I assume this change is because of the issues with `SliceHeader`/`StringHeader` in tinygo?

An alternative solution here would be to just pull the bytes from the buffer directly:

```go
if a.array.data.buffers[2] != nil {
    return a.array.data.buffers[2].Bytes()[a.offsets[beg]:a.offsets[end]]
}
return nil
```

This way we preserve the fact that we're not copying the data.

Reply:
Yes we changed this because of the `SliceHeader` issue.

When would this cause a copy? Aren't we slicing the string which is referencing the same underlying data?

Reply:
Strings in Go are immutable, so unless the compiler can confirm that `[]byte(str)` is only used as a temporary it will do a copy of the bytes to construct the byte slice from the string.